public/scripts/scripts.js

```diff
@@ -42,25 +40,26 @@ function getPlaylistTracks(access_token, allCallSongs, request_url, playListTrac
       console.log(tracks);
       tracks.map(function(track){
         playListTracks.push(track);
-        writePlayListToPanel(track, playListTracks);
       })
       console.log(playListTracks);
       if(response.next) {
         getPlaylistTracks(access_token, allCallSongs, response.next, playListTracks);
+      } else {
```

Comment:
i moved the `initializePlayListControl()` call here from the `$(document).ajaxComplete()` call. `$(document).ajaxComplete()` is weird, i don't think it's what you want here. it seems to me like it attaches an event listener that gets called from then on for _any_ ajax response that comes back, which would likely lead to weird behavior in the future if you add more ajax calls to this program.

this `else` statement gets hit if there aren't any more playlist tracks requests to make, and i think that's the behavior you were looking for. i also moved the `writePlayListToPanel(playListTracks)` down here from a few lines up. i think it makes more sense to call `writePlayListToPanel(playListTracks)` here, when we're 100% sure that playListTracks is completely filled and good to go. calling it farther up made it so that it would sometimes be called on a partially-filled `playListTracks`, and that seems like a recipe for weird bugs.

Reply:
Yeah, I was starting to notice that the lines housed inside .ajaxComplete() were firing, but wasn't 100% on whether that was because there was a listener on the document, or if it was due to getPLT firing unintentionally.